src/main/drivers/sonar_hcsr04.c

```diff
@@ -38,6 +41,17 @@
  */
 
 #if defined(SONAR)
+
+#if !defined(SONAR_TRIGGER_PIN) || !defined(SONAR_ECHO_PIN)
```

Comment:
Sorry, my fault.

When target does not define defaults, you may use `NONE` for sonar pins as default config, instead of reporting error. User can change it when sonar is attached.